src/components/common/WalletProvider/index.tsx

```diff
@@ -7,19 +10,27 @@ const WalletProvider = ({ children }: { children: ReactNode }): ReactElement =>
   const onboard = useOnboard()
   const onboardWallets = onboard?.state.get().wallets || []
   const [wallet, setWallet] = useState<ConnectedWallet | null>(getConnectedWallet(onboardWallets))
+  const chainId = useChainId()
+  const { safeAddress } = useSafeInfo()
 
   useEffect(() => {
     if (!onboard) return
 
-    const walletSubscription = onboard.state.select('wallets').subscribe((wallets) => {
+    const walletSubscription = onboard.state.select('wallets').subscribe(async (wallets) => {
       const newWallet = getConnectedWallet(wallets)
+
+      if (newWallet && chainId && safeAddress) {
+        const delegates = await getDelegates(chainId, { safe: safeAddress })
+        newWallet.isDelegate = delegates.results.some((delegate) => delegate.delegate === newWallet.address)
+      }
```

Comment:
This would delay wallet connection for all users to serve a very minor portion of users. I would not add isDelegate to the wallet object and instead use a separate hook.